MediaBrowser.Controller/Entities/MediaSourceWidthComparator.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.Intrinsics.X86;
+using MediaBrowser.Model.Dto;
+
+namespace MediaBrowser.Controller.Entities;
+
+/// <summary>
+/// Compare MediaSource of the same file by Video width <see cref="IComparer{T}" />.
+/// </summary>
+public class MediaSourceWidthComparator : IComparer<MediaSourceInfo>
+{
+    /// <inheritdoc />
+    public int Compare(MediaSourceInfo? x, MediaSourceInfo? y)
+    {
+        if (x is null && y is null)
+        {
+            return 0;
+        }
+
+        if (x is null)
+        {
+            return -1;
+        }
+
+        if (y is null)
+        {
+            return 1;
+        }
+
+        if (string.Equals(x.Path, y.Path, StringComparison.OrdinalIgnoreCase))
```

Comment:
If `Path` is an absolute path, we will never get the same paths for 2 different items. 🤷‍♂️

Reply:
Some containers like Matroska can have multiple video streams in the same file with the same path, and I’m unsure what would happen if we simply remove that part, as it’s been in place since the Emby era. This sorting wasn’t initially added for multi-version handling, but I’m unsure of its original intent. I’m not confident in removing it entirely, as it could potentially break things.

Reply:
Also I could imagine that some LiveTV plugins can supply HLS streams with multiple resolution in one MediaSource and having the same URL, which probably relies on this currently.

Reply:
> Some containers like Matroska can have multiple video streams in the same file with the same path

Oh, those. AFAIK, we don't support multiple video streams. But if it will be implemented in the future, that's fine.